### PR TITLE
Remove react foundation components

### LINKED ISSUE #2303: Top level dropdowns do not accept theme-ing.
Condenser's current top-level dropdowns exist outside the theming context. They cannot respond to 'night mode' theme changes (or any others for that matter)

Though theme-dark is set, the dropdown exists outside the regular tree.
screen shot 2018-01-08 at 2 51 19 pm

Two possible solutions (of many) add the theme class to either the `<body>` tag, or the root of the tree that holds the dropdown.

body:
screen shot 2018-01-08 at 2 53 06 pm
dropdown root:
screen shot 2018-01-08 at 2 50 52 pm

### PR DESCRIPTION
**Pull request by gl2748**
# Before:
before_real

# After:
after

The react-foundation-components library is a bug-bear, inconsistently maintained and unpredictable.
We'd rather not use it, but unfortunately it is used extensively our codebase.

This PR zaps the react-foundation-components dependency aiming for parity where possible, and subject to review, with a couple beneficial changes - namely:

- Dropdowns styled for night mode. Fixes #2303
- 'Flag' actions are consistently presented in a modal format - described here: https://github.com/steemit/condenser/issues/2258#issuecomment-355985057

# Demonstration of Flagging interaction change - no more dropdown...:
## Before.
b4_flag

## After.
after_flag2

ToDo.
- [ ] Look through the wording for the flag action / modal text.
- [ ] Remove Steemit maintained fork of https://github.com/steemit/foundation-sites.
- [x] PR review feedback.

### CODE REVIEW

File: src/app/components/pages/PickAccount.jsx [hunk 1]
```diff
@@ -285,7 +284,11 @@ class PickAccount extends React.Component {
                         style={{ maxWidth: '36rem', margin: '0 auto' }}
                     >
                         <br />
-                        <Progress tabIndex="0" value={10} max={100} />
+                        <progress max="100" value="10">
```
Comment (bnchdrff) [inline src/app/components/pages/PickAccount.jsx]: why use `<progress>` at all? we'll have to do custom styles for each browser anyways (only webkit styles are included in this PR); we should just do the `div.progress` i think

screenshot from 2018-01-25 09-46-44

Comment (gl2748) [reply src/app/components/pages/PickAccount.jsx]: That's what this is, it's html5 <progress> element, not a component...

Comment (bnchdrff) [reply src/app/components/pages/PickAccount.jsx]: it'll get native styling, and is ugly though (see the win7 example)

here's chrome (looks nice)

screenshot from 2018-01-25 12-27-02


and here's ff (blue, weird)

screenshot from 2018-01-25 12-27-12

File: src/app/components/modules/TopRightMenu.scss [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
     li:last-child {
-        padding: 0 0 0 1rem;
+    }
+}
+
+.Header__usermenu {
+    li {
+        a {
+            padding: 0.7rem 1rem;
+        }
```
Comment (bnchdrff) [inline src/app/components/modules/TopRightMenu.scss]: master:

screenshot from 2018-01-25 12-30-15


this branch:

screenshot from 2018-01-25 12-25-12

Comment (gl2748) [reply src/app/components/modules/TopRightMenu.scss]: Will fix.

**Comment by gl2748** [comment]
### Progress bar.

## Chrome:

screenshot from 2018-01-30 15-18-24

## FireFox

screenshot from 2018-01-30 15-19-25

**Comment by gl2748** [comment]
Rebased -> #2434 2434

**Pull request closed without merging**